src/bun.js/node/node_perf_hooks_histogram_binding.zig

```diff
@@ -0,0 +1,234 @@
+            const maxPercentileValue = this.hdrHist.value_at_percentile(100) orelse return .undefined;
+            var percent: f64 = 0;
+            var stack_allocator = std.heap.stackFallback(4096, bun.default_allocator);
+            var doubles = std.ArrayList(f64).init(stack_allocator.get());
+            defer doubles.deinit();
+
+            while (true) {
+                if (this.hdrHist.value_at_percentile(percent)) |val| {
+                    doubles.appendSlice(&.{ percent, @bitCast(val) }) catch |err| {
+                        globalObject.throwError(err, "failed to append to array");
+                        return .undefined;
+                    };
+                    if (val >= maxPercentileValue) {
+                        break;
+                    }
+                }
+                percent += ((100 - percent) / 2);
+            }
+
+            doubles.appendSlice(&.{ 100, @bitCast(maxPercentileValue) }) catch |err| {
+                globalObject.throwError(err, "failed to append max value to array");
+                return .undefined;
+            };
+
+            return Bun__createMapFromDoubleUint64TupleArray(globalObject, @as([*]const f64, @ptrCast(doubles.items)), doubles.items.len);
+        }
+    };
+    pub const percentiles = @as(PropertyGetter, percentiles_fn.callback);
+    pub const percentilesBigInt = getterAsFn(percentiles_fn.callback);
+
+    //
+    // additional functions
+
+    // record duration in nanoseconds
+    pub fn record(this: *This, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        const args = callframe.arguments(1).slice();
+        if (args.len != 1) {
+            globalThis.throwInvalidArguments("Expected 1 argument", .{});
+            return .zero;
+        }
+        const value = args[0].to(u64);
+        this.hdrHist.record_value(value, 1);
+        return .undefined;
+    }
+
+    // record time since last call to recordDelta
+    pub fn recordDelta(this: *This, _: *JSC.JSGlobalObject, _: *JSC.CallFrame) callconv(.C) JSValue {
+        if (this.delta_start) |start| {
+            const end = bun.timespec.now();
+            const diff = end.duration(&start);
+            this.hdrHist.record_value(@intCast(diff.nsec), 1);
+            this.delta_start = end;
+            return .undefined;
+        }
+
+        // first call no-ops
+        this.delta_start = bun.timespec.now();
+
+        return .undefined;
+    }
+
+    pub fn reset(this: *This, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        _ = globalThis;
+        _ = callframe;
+        this.hdrHist.reset();
+        return .undefined;
+    }
+
+    pub fn add(this: *This, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        const args = callframe.arguments(1).slice();
+        if (args.len != 1) {
```

Comment:
Generally, in JS you ignore it when there are too many arguments

```suggestion
        if (args.len < 1) {
```

Reply:
fixed in upcoming push